File: backend/schemas/investment_report.py
from pydantic import BaseModel


class InvestmentReportResponse(BaseModel):
    symbol: str

    # AI Summary
    summary: str

    # AI Recommendation
    recommendation: str
    confidence: int

    # Overall Rating
    score: float
    grade: str

    # Individual Scores
    valuation_score: float
    growth_score: float
    profitability_score: float
    liquidity_score: float
    leverage_score: float
    risk_score: float
    margin_of_safety: float
    # DCF

    intrinsic_value: float
    # AI Insights
    strengths: list[str]
    risks: list[str]

File: atlas-finance-ui/src/types/investmentReport.ts
export interface InvestmentReport {
  symbol: string;

  company_name: string;
  exchange?: string;
  sector?: string;
  industry?: string;
  country?: string;
  currency?: string;
  market_cap?: number;
  logo?: string;

  recommendation: string;
  confidence: number;
  score: number;
  grade: string;

  intrinsic_value: number;

  valuation_score: number;
  growth_score: number;
  profitability_score: number;
  liquidity_score: number;
  leverage_score: number;
  risk_score: number;

  summary: string;

  strengths: string[];
  risks: string[];
}